src/keywords/optionalint.cpp

```diff
@@ -55,9 +56,28 @@ std::string OptionalIntegerKeyword::textWhenNull() const { return textWhenNull_;
 // Deserialise from supplied LineParser, starting at given argument offset
 bool OptionalIntegerKeyword::deserialise(LineParser &parser, int startArg, const CoreData &coreData)
 {
+    std::optional<int> newValue;
+    std::string_view newString;
+    try
+    {
+        newValue = parser.argi(startArg);
+    }
+    catch (...)
+    {
+        newString = parser.argsv(startArg);
+        if (DissolveSys::stob(newString) == false)
+        {
+            if (newString != textWhenNull_)
+                Messenger::error("Unkonwn input, {}, provided\n", newString);
+            else
+                newString = textWhenNull_;
+        }
+        else
+            DissolveSys::stob(newString);
```

Comment:
The logic is a bit convoluted here, and after thinking about it I don't think we need to consider the `stob()` part.  I'd suggest removing the `newString` variable and doing something like:
```suggestion
        // Check explicitly for the null text
        if (DissolveSys::sameString(parser.argv(startArg), textWhenNull_))
            setData(std::nullopt);
        else
                return Messenger::error("Unknown input, {}, provided\n", newString);
```